### PR TITLE
[Google Blockly] Wrap FieldDropdown constructor

### PR DESCRIPTION
**Pull request by ajpal**
CDO Blockly FieldDropdown constructor:
`Blockly.FieldDropdown = function(menuGenerator, opt_changeHandler, opt_alwaysCallChangeHandler) {...}`
The `opt_changeHandler` callback is called whenever a value is selected from the dropdown. 

Google Blockly FieldDropdown constructor:
`Blockly.FieldDropdown = function(menuGenerator, opt_validator, opt_config) {...}`
The `opt_validator` callback is called whenever the selected option changes (for any reason, whether or not the change was triggered by a user action).

Usage in Flappy:
```
function onSoundSelected(soundValue) {
  if (soundValue === RANDOM_VALUE) {
    return;
  }
  studioApp().playAudio(utils.stripQuotes(soundValue));
}
var soundDropdown = new blockly.FieldDropdown(this.VALUES, onSoundSelected);
```

The sound is only supposed to play when the user selects an option from the dropdown (the sound just plays once as a preview so the user knows what it will sound like at runtime). However, since the callback was now being treated as a validator function, the sound was playing many many times (basically every time the user interacted with the block, not just when they select a new value).

This is in part due to a new feature in Google Blockly called insertion markers or shadow blocks (the gray block showing where the playSound block will fall): 
Sep-01-2020 12-30-53
Shadow blocks are created when you start dragging a block and deleted when you stop. They call through the same initialization path as the real block. So we were hitting the validator function every time you drag the block (since it creates a shadow block and sets the value of it), resulting in the sound being played every time you interact with the block, not just when a new value is chosen.

The solution is to add a layer in the `GoogleBlocklyWrapper` wrapping the FieldDropdown constructor. This way, we can make sure we only call the callback when something has actually changed. 

- [jira](https://codedotorg.atlassian.net/browse/STAR-1200)

### CODE REVIEW

File: apps/src/sites/studio/pages/googleBlocklyWrapper.js [hunk 1]
```diff
@@ -87,13 +86,34 @@ function initializeBlocklyWrapper(blocklyInstance) {
   blocklyWrapper.wrapReadOnlyProperty('Procedures');
   blocklyWrapper.wrapReadOnlyProperty('removeChangeListener');
   blocklyWrapper.wrapReadOnlyProperty('RTL');
+  blocklyWrapper.wrapReadOnlyProperty('selected');
```
Comment (ajpal) [inline apps/src/sites/studio/pages/googleBlocklyWrapper.js:89]: This is not really related to FieldDropdowns, just a property I forgot to wrap in the first PR, since it's not used in apps code anywhere. It's very useful during development though:
image

**Review by breville — APPROVED** [review]
Nice work.  As discussed separately, it will be interesting to see if this solution works more broadly for other uses of the change handler when going beyond Flappy.

**Comment by ajpal** [comment]
> It will be interesting to see if this solution works more broadly for other uses of the change handler when going beyond Flappy.

I did a quick search in the rest of the codebase, and at least for dropdown fields, the only other use of the change handler parameter is the play sound block in Play Lab. Didn't investigate further, but I assume the use case is similar/the same there.
